internal/daemon/daemon.go

```diff
@@ -622,11 +652,87 @@ func (d *Daemon) StartAPI(bootstrap bool, initConfig map[string]string, newConfi
 	return nil
 }
 
+// UpdateServers updates and start/stops the additional listeners.
+func (d *Daemon) UpdateServers() error {
+	var activeExtensionServers []string
+	for name := range d.config.GetServers() {
+		activeExtensionServers = append(activeExtensionServers, name)
+	}
+
+	// Stop all additional listeners which got removed from the config.
+	for _, serverName := range d.ExtensionServers() {
+		if !shared.ValueInSlice(serverName, activeExtensionServers) {
+			// Remove their config.
+			d.extensionServersMu.Lock()
+			extensionServer, ok := d.extensionServers[serverName]
+			if !ok {
+				// There isn't any additional listener set that matches this name.
+				d.extensionServersMu.Unlock()
+				continue
+			}
+
+			// Set an empty config for this specific additional listener.
+			extensionServer.ServerConfig = types.ServerConfig{}
+
+			// Reassign the map struct.
+			d.extensionServers[serverName] = extensionServer
+			d.extensionServersMu.Unlock()
+
+			err := d.endpoints.DownByName(serverName)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// Restart all additional listeners which received an update.
+	for serverName, extensionServerConfig := range d.config.GetServers() {
```

Comment:
If you instantiate `GetServers()` once, you won't have to do the map copying twice.

Reply:
Fixed.